vendorconsent/tcf2/metadata.go

```diff
@@ -125,6 +126,10 @@ func (c ConsentMetadata) VendorListVersion() uint16 {
 	return binary.BigEndian.Uint16([]byte{leftByte, rightByte})
 }
 
+func (c ConsentMetadata) MaxLegitimateInterestVendorID() uint16 {
```

Comment:
This has a strong correlation to the `VendorLegitInterest` method and I don't think that's captured in this name. How about `VendorLegitInterestMaxID`?